OSDOSC-7864: Migrated Using Group Sync Operator with Okta and ROSA (#65058)
openshift/openshift-docs: rosa_tutorials/rosa-mobb-groupsync-okta.adoc

:_content-type: ASSEMBLY
[id="rosa-mobb-groupsync-okta"]
= Tutorial: Using Group Sync Operator with Okta and ROSA
include::_attributes/attributes-openshift-dedicated.adoc[]
:context: rosa-mobb-groupsync-okta

toc::[]

//Mobb content metadata
//Brought into ROSA product docs 2023-09-18
//---
//date: '2023-07-25
//title: Using Group Sync Operator with Okta and ROSA
//authors:
//  - Thatcher Hubbard
//---

> This guide focuses on how to synchronize Identity Provider (IDP) groups and users after configuring authentication in OpenShift Cluster Manager (OCM).

Reviewer: ```suggestion
This guide focuses on how to synchronize Identity Provider (IDP) groups and users after configuring authentication in Red Hat OpenShift Cluster Manager (OCM).
```

To set up group synchronization from Okta to ROSA you must:

. Define groups and assign users in Okta.

> . Install the Group Sync Operator from the OpenShift Operator Hub.

Reviewer: ```suggestion
. Install the Group Sync Operator from the OpenShift OperatorHub.
```

. Create and configure a new Group Sync instance.
. Set a synchronization schedule.
. Test the synchronization process.

[id="rosa-mobb-groupsync-okta-define-groups"]
== Define groups and assign users in Okta

To synchronize groups and users with ROSA, they must exist in Okta.

. Create the groups to synchronize with ROSA if they do not already exist:
+
image::okta-grp.png[Okta Groups]

. Create user IDs to synchronize with ROSA if they do not already exist:
+
image::okta-usr.png[Okta Users]

. Assign the newly created users to the appropriate group:
+
image::okta-assign.png[Okta add user to group]

[id="rosa-mobb-groupsync-okta-install-group-sync"]

> == Install the Group Sync Operator from the OpenShift Operator Hub

Reviewer: ```suggestion
== Install the Group Sync Operator from the OpenShift OperatorHub
```

. In the OpenShift Operator Hub, select the **Group Sync Operator**:
+

> image::grp-sync-opr-hub.png[Group Sync in Operator Hub]

Reviewer: ```suggestion
image::grp-sync-opr-hub.png[Group Sync in OperatorHub]
```

====

Ideally, a user would be created inside the Okta organization that was specifically for group synchronizations, which should only need to be able to read groups. Creating a user with 'Read Administrator' permissions on the account would be a good place to start for following the principle of "least privilege." That user can then issue a token that includes only those permissions.

[id="rosa-mobb-groupsync-okta-access-token"]
== Create an Okta API Access Token

* Log in as a user that has minimally has Group and User read permissions and generate an API token in Okta:
+
image::okta-api-token.png[Okta API token creation]

[id="rosa-mobb-groupsync-okta-group-sync-instance"]
== Create and configure a new Group Sync instance

. Create a new secret named `okta-group-sync` in the **group-sync-operator** namespace. This will contain the Okta API key that was just created.

. Using the OpenShift CLI (`oc`), create the secret using the following format:
+
[source,terminal]
----
$ oc create secret generic okta-api-token --from-literal='okta-api-token=${API_TOKEN}' -n group-sync-operator
----

> . Obtain values from Okta for the AppId and URL. The AppId is the client ID under the application created to support OpenID for the OCP Cluster(s). The URL is the same as the one used to admin Okta, without the `-admin` in the first term and should look something like this:

Reviewer: ```suggestion
. Obtain values from Okta for the AppId and URL. The AppId is the client ID under the application created to support OpenID for the {product-name-first} cluster(s). The URL is the same as the one used to admin Okta, without the `-admin` in the first term and should look something like this:
```
